src/Avalonia.Controls/Button.cs

```diff
@@ -102,6 +105,9 @@ static Button()
             AccessKeyHandler.AccessKeyPressedEvent.AddClassHandler<Button>((lbl, args) => lbl.OnAccessKey(args));
```

Comment:
I'm going to slip this question in here:

I noticed button is doing the following to detect property changes in the static constructor:

```csharp
  CommandProperty.Changed.Subscribe(CommandChanged);
  CommandParameterProperty.Changed.Subscribe(CommandParameterChanged);
  IsDefaultProperty.Changed.Subscribe(IsDefaultChanged);
  IsCancelProperty.Changed.Subscribe(IsCancelChanged);
```

However, Avalonia has an overridable method `OnPropertyChanged` that can be used internally. This is what is done for `SplitButton`.

Which technique is more correct/current? My guess is the `OnPropertyChanged` overridable method was added later and `Button` just has very old code here. I have to think this is significantly more overhead to handle things by subscribing to a property changed event as well.

Reply:
OnPropertyChanged should be preferred, yes. Though overhead should be minimal now, Changed observable was optimized heavily before.
Changed observable is still needed, when you need to subscribe on attached properties without access to the OnPropertyChanged method.

Reply:
I always use the `OnPropertyChanged` override as it avoids the allocations with the subscriptions and I think its cleaner. I think I've seen some comments from the Avalonia team that using `OnPropertyChanged` is the better way to do it now as it avoid the allocation and I think its actually faster. I did try once to change everything in the controls, but its used in so many places I decided it wasn't worth it.

Reply:
Ok, makes sense. I'm going to slip in a quick update to button. It will simplify a few things while I'm looking at it.

Reply:
This is resolved with commits I'll push out later. I updated button and doing so caught a bug in SplitButton as well. Now they both use OnPropertyChanged internally instead of subscriptions.

Reply:
All Button.cs changes were moved to another PR: https://github.com/AvaloniaUI/Avalonia/pull/7451. This change was made there. Marking resolved here.